<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="rajas-imac.lan" failures="51" tests="303" name="Default test" time="4309.723" errors="0" timestamp="2025-04-21T15:00:08 PDT">
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="122.472"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="122.474"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="135.041"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="135.051"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="146.243"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="146.579"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="150.524"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="150.539"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="150.552"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="151.377"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="184.149"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="176.561"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="220.959"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="255.441"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="232.844"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="232.64"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="247.091"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="231.959"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="231.971"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="268.55"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="210.199"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="213.385"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="177.496"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="177.53"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="184.48"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="201.458"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="209.23"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="209.243"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="202.203">
    <failure type="java.lang.AssertionError" message="Test case Failed: Out put from console is not matched with the expected output expected [] but found [Submission Successful]">
      <![CDATA[java.lang.AssertionError: Test case Failed: Out put from console is not matched with the expected output expected [] but found [Submission Successful]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.ArrayStepDef.the_user_is_not_able_to_see_Submission_success_message(ArrayStepDef.java:336)
at ✽.The user is not able to see success message Submission successful(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Arrays.feature:186)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="212.199">
    <failure type="org.openqa.selenium.NoSuchElementException" message="no such element: Unable to locate element: {&amp;quot;method&amp;quot;:&amp;quot;xpath&amp;quot;,&amp;quot;selector&amp;quot;:&amp;quot;//pre[text()=&amp;quot;Error occurred during submission&amp;quot;]&amp;quot;}
  (Session info: chrome=135.0.7049.96)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Build info: version: &amp;apos;4.31.0&amp;apos;, revision: &amp;apos;1ef9f18787*&amp;apos;
System info: os.name: &amp;apos;Mac OS X&amp;apos;, os.arch: &amp;apos;x86_64&amp;apos;, os.version: &amp;apos;14.6.1&amp;apos;, java.version: &amp;apos;23&amp;apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [180c0d8ace81bf614ac3eb7a7be3c007, findElement {using=xpath, value=//pre[text()=&amp;quot;Error occurred during submission&amp;quot;]}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:51040}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:51040/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 180c0d8ace81bf614ac3eb7a7be3c007">
      <![CDATA[org.openqa.selenium.NoSuchElementException: no such element: Unable to locate element: {"method":"xpath","selector":"//pre[text()="Error occurred during submission"]"}
  (Session info: chrome=135.0.7049.96)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Build info: version: '4.31.0', revision: '1ef9f18787*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.6.1', java.version: '23'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [180c0d8ace81bf614ac3eb7a7be3c007, findElement {using=xpath, value=//pre[text()="Error occurred during submission"]}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:51040}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:51040/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 180c0d8ace81bf614ac3eb7a7be3c007
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:501)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:485)
at org.openqa.selenium.remote.ErrorCodec.decode(ErrorCodec.java:167)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:138)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:50)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:215)
at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:216)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:174)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElement(ElementLocation.java:165)
at org.openqa.selenium.remote.ElementLocation.findElement(ElementLocation.java:59)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:368)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:362)
at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:68)
at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38)
at jdk.proxy2/jdk.proxy2.$Proxy44.getText(Unknown Source)
at com.dsalgo.automation.pages.ArrayPage.verifyOutputError(ArrayPage.java:118)
at com.dsalgo.automation.stepdefinations.ArrayStepDef.the_user_see_an_error_message(ArrayStepDef.java:320)
at ✽.The user see an error message Error occurred during submission(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Arrays.feature:180)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="208.793"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="219.772"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="192.983"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="164.287"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="199.237"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="199.72"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="164.822">
    <failure type="java.lang.AssertionError" message="Test case Failed: Out put from console is not matched with the expected output expected [] but found [Submission Successful]">
      <![CDATA[java.lang.AssertionError: Test case Failed: Out put from console is not matched with the expected output expected [] but found [Submission Successful]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.ArrayStepDef.the_user_is_not_able_to_see_Submission_success_message(ArrayStepDef.java:336)
at ✽.The user is not able to see success message Submission successful(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Arrays.feature:228)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="178.569"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="193.787"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="200.741"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="146.926">
    <failure type="org.openqa.selenium.WebDriverException" message="Timed out waiting for driver server to shutdown.
Build info: version: &amp;apos;4.31.0&amp;apos;, revision: &amp;apos;1ef9f18787*&amp;apos;
System info: os.name: &amp;apos;Mac OS X&amp;apos;, os.arch: &amp;apos;x86_64&amp;apos;, os.version: &amp;apos;14.6.1&amp;apos;, java.version: &amp;apos;23&amp;apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [0e4aed5a208dfa20c08e44adf39c2e46, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:51340}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:51340/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 0e4aed5a208dfa20c08e44adf39c2e46">
      <![CDATA[org.openqa.selenium.WebDriverException: Timed out waiting for driver server to shutdown.
Build info: version: '4.31.0', revision: '1ef9f18787*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.6.1', java.version: '23'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [0e4aed5a208dfa20c08e44adf39c2e46, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:51340}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:51340/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 0e4aed5a208dfa20c08e44adf39c2e46
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:312)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:161)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:618)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:622)
at org.openqa.selenium.remote.RemoteWebDriver.quit(RemoteWebDriver.java:446)
at org.openqa.selenium.chromium.ChromiumDriver.quit(ChromiumDriver.java:418)
at com.dsalgo.automation.driver.DriverFactory.quitDriver(DriverFactory.java:60)
at com.dsalgo.automation.hooks.Hooks.tearDownFreshBrowser(Hooks.java:50)
Caused by: org.openqa.selenium.net.UrlChecker$TimeoutException: Timed out waiting for http://localhost:2732/shutdown to become unavailable after 3849 ms
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:151)
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:303)
... 8 more
Caused by: java.util.concurrent.TimeoutException
at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:144)
... 9 more
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="148.24"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="143.817"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="147.794"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="150.938"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="148.32"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="155.146"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="157.485"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="142.559"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="137.563">
    <failure type="java.lang.AssertionError" message="Test Case Failed : Expected output is not matching with Actual output expected [ No tests were collected] but found [No tests were collected]">
      <![CDATA[java.lang.AssertionError: Test Case Failed : Expected output is not matching with Actual output expected [ No tests were collected] but found [No tests were collected]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.ArrayStepDef.the_user_is_able_to_see_error_message_No_tests_were_Collected(ArrayStepDef.java:500)
at ✽.The user should be able to see error message No tests were Collected(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Arrays.feature:306)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="102.091"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="105.071"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="109.323"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="116.468"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="113.898"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="131.557"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="136.863">
    <failure type="java.lang.AssertionError" message="Test Case Failed : Expected output is not matching with Actual output expected [ No tests were collected] but found [No tests were collected]">
      <![CDATA[java.lang.AssertionError: Test Case Failed : Expected output is not matching with Actual output expected [ No tests were collected] but found [No tests were collected]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.ArrayStepDef.the_user_is_able_to_see_error_message_No_tests_were_Collected(ArrayStepDef.java:500)
at ✽.The user should be able to see error message No tests were Collected(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Arrays.feature:324)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="110.589"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="109.119"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="138.481">
    <failure type="java.lang.AssertionError" message="Test Case Failed : Expected and Actual results are not matching expected [ No tests were collected] but found [Submission Successful]">
      <![CDATA[java.lang.AssertionError: Test Case Failed : Expected and Actual results are not matching expected [ No tests were collected] but found [Submission Successful]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.ArrayStepDef.the_user_is_not_able_to_see_Submission_success_message_for_SquaresofSortedArray_code(ArrayStepDef.java:464)
at ✽.The user is not able to see success message Submission successful for Squares of a Sorted Array code(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Arrays.feature:312)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="73.493"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="70.006"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="82.736"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="73.317"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="97.439"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="113.537"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="123.879"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="139.768"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="149.789"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="155.109"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="95.301">
    <failure type="java.lang.AssertionError" message="&amp;quot;Test failed:Page title is not matched with Actual Title expected [doubly linked list] but found [www.javatpoint.com]">
      <![CDATA[java.lang.AssertionError: "Test failed:Page title is not matched with Actual Title expected [doubly linked list] but found [www.javatpoint.com]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.LinkedListStepDef.the_user_should_be_redirected_to_doubly_Linked_List_page(LinkedListStepDef.java:181)
at ✽.The user should be redirected to doubly linked list page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/LinkedList.feature:114)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="104.567">
    <failure type="java.lang.AssertionError" message="&amp;quot;Test failed:Page title is not matched with Actual Title expected [singly linked list] but found [www.javatpoint.com]">
      <![CDATA[java.lang.AssertionError: "Test failed:Page title is not matched with Actual Title expected [singly linked list] but found [www.javatpoint.com]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.LinkedListStepDef.the_user_should_be_redirected_to_singly_Linked_List_page(LinkedListStepDef.java:152)
at ✽.The user should be redirected to singly linked list page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/LinkedList.feature:102)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="101.36">
    <failure type="java.lang.AssertionError" message="&amp;quot;Test failed:Page title is not matched with Actual Title expected [linked list] but found [www.javatpoint.com]">
      <![CDATA[java.lang.AssertionError: "Test failed:Page title is not matched with Actual Title expected [linked list] but found [www.javatpoint.com]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.LinkedListStepDef.the_user_should_be_redirected_to_Linked_List_page_through_Linkedlist_hyperlink(LinkedListStepDef.java:166)
at ✽.The user should be redirected to linked list link page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/LinkedList.feature:108)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="88.392">
    <failure type="org.openqa.selenium.SessionNotCreatedException" message="Could not start a new session. Response code 500. Message: session not created: DevToolsActivePort file doesn&amp;apos;t exist 
Host info: host: &amp;apos;rajas-imac.lan&amp;apos;, ip: &amp;apos;192.168.86.24&amp;apos;
Build info: version: &amp;apos;4.31.0&amp;apos;, revision: &amp;apos;1ef9f18787*&amp;apos;
System info: os.name: &amp;apos;Mac OS X&amp;apos;, os.arch: &amp;apos;x86_64&amp;apos;, os.version: &amp;apos;14.6.1&amp;apos;, java.version: &amp;apos;23&amp;apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [null, newSession {capabilities=[Capabilities {browserName: chrome, goog:chromeOptions: {args: [], binary: /Applications/Google Chrome..., extensions: []}}]}]">
      <![CDATA[org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: session not created: DevToolsActivePort file doesn't exist 
Host info: host: 'rajas-imac.lan', ip: '192.168.86.24'
Build info: version: '4.31.0', revision: '1ef9f18787*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.6.1', java.version: '23'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [null, newSession {capabilities=[Capabilities {browserName: chrome, goog:chromeOptions: {args: [], binary: /Applications/Google Chrome..., extensions: []}}]}]
at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:114)
at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:75)
at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:61)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:187)
at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:216)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:174)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:245)
at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:174)
at org.openqa.selenium.chromium.ChromiumDriver.<init>(ChromiumDriver.java:110)
at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:88)
at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:83)
at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:72)
at com.dsalgo.automation.driver.DriverFactory.initializeDriver(DriverFactory.java:42)
at com.dsalgo.automation.hooks.Hooks.setUpFreshBrowser(Hooks.java:33)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="97.05">
    <failure type="java.lang.AssertionError" message="Test Case Failed: Title of the page is not matched with Expected Title expected [www.javatpoint.com] but found [type of the linked list]">
      <![CDATA[java.lang.AssertionError: Test Case Failed: Title of the page is not matched with Expected Title expected [www.javatpoint.com] but found [type of the linked list]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.LinkedListStepDef.the_user_should_be_redirected_to_typesof_Linked_List_page_through_link(LinkedListStepDef.java:196)
at ✽.The user should be redirected to type of the linked list page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/LinkedList.feature:120)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="96.068"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="100.23"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="108.575"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="123.192"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="133.565"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="150.864"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="125.043"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="125.036"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="160.472"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="124.733"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="91.032"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="131.252"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="115.299"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="117.571"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="144.177"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="121.461"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="121.69"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="150.566"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="125.934"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="129.846"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="87.703"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="105.059"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="107.033"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="96.267"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="110.037"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="99.044">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Queue.feature:17)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="123.412"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="152.522"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="125.534"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="123.697"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="52.39"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="54.072">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Queue.feature:47)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="72.459"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="87.425"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="87.419"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="87.447"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="79.895"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="89.454"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="95.049"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="95.045"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="85.95"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="88.43">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Queue.feature:77)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="89.827"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="60.955"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="79.843"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="104.727"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="106.302"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="89.808">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Queue.feature:107)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="90.793"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="104.304"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="75.352"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="85.066"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="85.326"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="75.933"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="72.809"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="85.079"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="86.674"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="78.23">
    <failure type="java.lang.AssertionError" message="Test failed: Incorrect error message displayed.">
      <![CDATA[java.lang.AssertionError: Test failed: Incorrect error message displayed.
at com.dsalgo.automation.stepdefinations.RegisterStepDef.then_the_correct_error_message_is_not_displayed(RegisterStepDef.java:145)
at ✽.The correct error message is not displayed(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Register.feature:35)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="69.725"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="80.374"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="107.711">
    <failure type="java.lang.AssertionError" message="Test failed: Incorrect error message displayed.">
      <![CDATA[java.lang.AssertionError: Test failed: Incorrect error message displayed.
at com.dsalgo.automation.stepdefinations.RegisterStepDef.then_the_correct_error_message_is_not_displayed(RegisterStepDef.java:145)
at ✽.The correct error message is not displayed(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Register.feature:40)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="86.662">
    <failure type="java.lang.AssertionError" message="Test failed: Incorrect error message displayed.">
      <![CDATA[java.lang.AssertionError: Test failed: Incorrect error message displayed.
at com.dsalgo.automation.stepdefinations.RegisterStepDef.then_the_correct_error_message_is_not_displayed(RegisterStepDef.java:145)
at ✽.The correct error message is not displayed(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Register.feature:50)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="86.69">
    <failure type="java.lang.AssertionError" message="Test failed: Incorrect error message displayed.">
      <![CDATA[java.lang.AssertionError: Test failed: Incorrect error message displayed.
at com.dsalgo.automation.stepdefinations.RegisterStepDef.then_the_correct_error_message_is_not_displayed(RegisterStepDef.java:145)
at ✽.The correct error message is not displayed(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Register.feature:55)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="94.376">
    <failure type="java.lang.AssertionError" message="Test failed: Incorrect error message displayed.">
      <![CDATA[java.lang.AssertionError: Test failed: Incorrect error message displayed.
at com.dsalgo.automation.stepdefinations.RegisterStepDef.then_the_correct_error_message_is_not_displayed(RegisterStepDef.java:145)
at ✽.The correct error message is not displayed(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Register.feature:45)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="77.107">
    <failure type="java.lang.AssertionError" message="expected [password_mismatch:The two password fields didn’t match.] but found [New Account Created. You are logged in as abc@testers.com]">
      <![CDATA[java.lang.AssertionError: expected [password_mismatch:The two password fields didn’t match.] but found [New Account Created. You are logged in as abc@testers.com]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at org.testng.Assert.assertEquals(Assert.java:639)
at com.dsalgo.automation.stepdefinations.RegisterStepDef.then_the_user_should_be_redirected_to_home_page_of_ds_algo_with_message(RegisterStepDef.java:210)
at ✽.The user should be redirected to Home Page of DS Algo with message "New Account Created. You are logged in as "(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Register.feature:65)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="109.072"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="109.069"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="111.239"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="80.147"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="87.126"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="85.172"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="103.419"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="103.604"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="103.709"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="103.969"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="115.095"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="101.943"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="133.794"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="133.24"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="104.284"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="129.18"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="129.19"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="111.822"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="114.458"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="114.757"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="97.95"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="99.042"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="101.656">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:21)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="134.74"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="113.401"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="115.213"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="117.911"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="123.828"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="146.047"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="117.589"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="126.982">
    <failure type="java.lang.AssertionError" message="expected [] but found [ ]">
      <![CDATA[java.lang.AssertionError: expected [] but found [ ]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at org.testng.Assert.assertEquals(Assert.java:639)
at com.dsalgo.automation.stepdefinations.TreeStepDef.then_the_user_should_be_able_to_view_all_the_data_in_table(TreeStepDef.java:89)
at ✽.The user should be able to view all the data in table(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:50)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="99.13">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:85)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="112.038"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="114.664"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="140.741">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:55)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="132.394"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="133.241"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="146.22"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="122.492"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="137.393"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="182.213"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="182.226"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="183.536"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="203.709">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:123)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="212.835"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="240.171"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="217.805"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="219.05"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="242.634"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="270.155"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="138.498"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="138.738"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="153.506">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:157)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="150.66"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="151.703"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="192.023"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="183.885"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="183.69"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="188.786"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="184.823">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:191)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="162.66"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="173.938"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="174.455">
    <failure type="org.openqa.selenium.WebDriverException" message="Timed out waiting for driver server to shutdown.
Build info: version: &amp;apos;4.31.0&amp;apos;, revision: &amp;apos;1ef9f18787*&amp;apos;
System info: os.name: &amp;apos;Mac OS X&amp;apos;, os.arch: &amp;apos;x86_64&amp;apos;, os.version: &amp;apos;14.6.1&amp;apos;, java.version: &amp;apos;23&amp;apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [eaddb0d9a045c7a999aeac5cf6b427a7, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:53536}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:53536/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: eaddb0d9a045c7a999aeac5cf6b427a7">
      <![CDATA[org.openqa.selenium.WebDriverException: Timed out waiting for driver server to shutdown.
Build info: version: '4.31.0', revision: '1ef9f18787*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.6.1', java.version: '23'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [eaddb0d9a045c7a999aeac5cf6b427a7, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:53536}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:53536/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: eaddb0d9a045c7a999aeac5cf6b427a7
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:312)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:161)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:618)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:622)
at org.openqa.selenium.remote.RemoteWebDriver.quit(RemoteWebDriver.java:446)
at org.openqa.selenium.chromium.ChromiumDriver.quit(ChromiumDriver.java:418)
at com.dsalgo.automation.driver.DriverFactory.quitDriver(DriverFactory.java:60)
at com.dsalgo.automation.hooks.Hooks.tearDownFreshBrowser(Hooks.java:50)
Caused by: org.openqa.selenium.net.UrlChecker$TimeoutException: Timed out waiting for http://localhost:9551/shutdown to become unavailable after 3379 ms
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:151)
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:303)
... 8 more
Caused by: java.util.concurrent.TimeoutException
at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:144)
... 9 more
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="116.881"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="132.637"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="168.136"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="231.486"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="209.604">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:229)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="210.324"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="161.275">
    <failure type="org.openqa.selenium.WebDriverException" message="Timed out waiting for driver server to shutdown.
Build info: version: &amp;apos;4.31.0&amp;apos;, revision: &amp;apos;1ef9f18787*&amp;apos;
System info: os.name: &amp;apos;Mac OS X&amp;apos;, os.arch: &amp;apos;x86_64&amp;apos;, os.version: &amp;apos;14.6.1&amp;apos;, java.version: &amp;apos;23&amp;apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [55a185b2c58bd9b61986084ee1246829, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:53698}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:53698/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 55a185b2c58bd9b61986084ee1246829">
      <![CDATA[org.openqa.selenium.WebDriverException: Timed out waiting for driver server to shutdown.
Build info: version: '4.31.0', revision: '1ef9f18787*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.6.1', java.version: '23'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [55a185b2c58bd9b61986084ee1246829, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:53698}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:53698/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 55a185b2c58bd9b61986084ee1246829
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:312)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:161)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:618)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:622)
at org.openqa.selenium.remote.RemoteWebDriver.quit(RemoteWebDriver.java:446)
at org.openqa.selenium.chromium.ChromiumDriver.quit(ChromiumDriver.java:418)
at com.dsalgo.automation.driver.DriverFactory.quitDriver(DriverFactory.java:60)
at com.dsalgo.automation.hooks.Hooks.tearDownFreshBrowser(Hooks.java:50)
Caused by: org.openqa.selenium.net.UrlChecker$TimeoutException: Timed out waiting for http://localhost:15191/shutdown to become unavailable after 3376 ms
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:151)
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:303)
... 8 more
Caused by: java.util.concurrent.TimeoutException
at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:144)
... 9 more
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="157.011">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:259)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="230.354"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="219.011"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="180.035"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="234.587"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="193.883"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="142.877"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="185.61">
    <failure type="org.openqa.selenium.WebDriverException" message="Timed out waiting for driver server to shutdown.
Build info: version: &amp;apos;4.31.0&amp;apos;, revision: &amp;apos;1ef9f18787*&amp;apos;
System info: os.name: &amp;apos;Mac OS X&amp;apos;, os.arch: &amp;apos;x86_64&amp;apos;, os.version: &amp;apos;14.6.1&amp;apos;, java.version: &amp;apos;23&amp;apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [eb5c6c14479b986334627017e24c5578, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:53784}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:53784/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: eb5c6c14479b986334627017e24c5578">
      <![CDATA[org.openqa.selenium.WebDriverException: Timed out waiting for driver server to shutdown.
Build info: version: '4.31.0', revision: '1ef9f18787*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.6.1', java.version: '23'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [eb5c6c14479b986334627017e24c5578, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:53784}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:53784/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: eb5c6c14479b986334627017e24c5578
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:312)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:161)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:618)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:622)
at org.openqa.selenium.remote.RemoteWebDriver.quit(RemoteWebDriver.java:446)
at org.openqa.selenium.chromium.ChromiumDriver.quit(ChromiumDriver.java:418)
at com.dsalgo.automation.driver.DriverFactory.quitDriver(DriverFactory.java:60)
at com.dsalgo.automation.hooks.Hooks.tearDownFreshBrowser(Hooks.java:50)
Caused by: org.openqa.selenium.net.UrlChecker$TimeoutException: Timed out waiting for http://localhost:46917/shutdown to become unavailable after 3219 ms
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:151)
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:303)
... 8 more
Caused by: java.util.concurrent.TimeoutException
at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:144)
... 9 more
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="178.846"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="177.926"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="174.394">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:293)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="157.162"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="157.182"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="157.166"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="162.197"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="202.634"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="192.175">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:323)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="185.505"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="196.338"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="161.754"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="180.828"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="180.84">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:353)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="186.426"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="208.21"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="214.908"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="187.138"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="198.992"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="170.74"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="164.538">
    <failure type="org.openqa.selenium.WebDriverException" message="Timed out waiting for driver server to shutdown.
Build info: version: &amp;apos;4.31.0&amp;apos;, revision: &amp;apos;1ef9f18787*&amp;apos;
System info: os.name: &amp;apos;Mac OS X&amp;apos;, os.arch: &amp;apos;x86_64&amp;apos;, os.version: &amp;apos;14.6.1&amp;apos;, java.version: &amp;apos;23&amp;apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [206c1a7810a5926c5c9097e601744dd2, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54086}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54086/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 206c1a7810a5926c5c9097e601744dd2">
      <![CDATA[org.openqa.selenium.WebDriverException: Timed out waiting for driver server to shutdown.
Build info: version: '4.31.0', revision: '1ef9f18787*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.6.1', java.version: '23'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [206c1a7810a5926c5c9097e601744dd2, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54086}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54086/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 206c1a7810a5926c5c9097e601744dd2
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:312)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:161)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:618)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:622)
at org.openqa.selenium.remote.RemoteWebDriver.quit(RemoteWebDriver.java:446)
at org.openqa.selenium.chromium.ChromiumDriver.quit(ChromiumDriver.java:418)
at com.dsalgo.automation.driver.DriverFactory.quitDriver(DriverFactory.java:60)
at com.dsalgo.automation.hooks.Hooks.tearDownFreshBrowser(Hooks.java:50)
Caused by: org.openqa.selenium.net.UrlChecker$TimeoutException: Timed out waiting for http://localhost:5467/shutdown to become unavailable after 3219 ms
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:151)
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:303)
... 8 more
Caused by: java.util.concurrent.TimeoutException
at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:144)
... 9 more
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="118.994"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="174.542">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:387)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="146.336"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="157.074"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="146.508"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="160.947"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="215.715"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="223.255"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="292.768"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="235.163">
    <failure type="java.lang.AssertionError" message="The Practice Questions page is blank.No content Found expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: The Practice Questions page is blank.No content Found expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertTrue(Assert.java:45)
at com.dsalgo.automation.stepdefinations.DataStructureStepDef.user_is_redirected_to_blank_practice_questions_page(DataStructureStepDef.java:150)
at ✽.the user is redirected to the blank page of "data structures-introduction" "Practice Questions"(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/datastructure.feature:20)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="315.186">
    <failure type="java.lang.AssertionError" message="Test failed: Page is Empty.">
      <![CDATA[java.lang.AssertionError: Test failed: Page is Empty.
at com.dsalgo.automation.stepdefinations.CommonStepDef.then_the_user_should_be_navigated_to_practice_questions_page(CommonStepDef.java:46)
at ✽.The User should be navigated to Practice Questions page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/Tree.feature:417)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="267.568"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="294.962"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="246.035"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="250.312">
    <failure type="java.lang.AssertionError" message="No error message displayed. expected [false] but found [true]">
      <![CDATA[java.lang.AssertionError: No error message displayed. expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertFalse(Assert.java:67)
at com.dsalgo.automation.stepdefinations.DataStructureStepDef.the_user_should_see_an_error_message(DataStructureStepDef.java:229)
at ✽.the user should see an error message(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/datastructure.feature:31)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="316.583"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="164.977"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="189.203">
    <failure type="java.lang.AssertionError" message=" User was not redirected to the Home page. expected [https://dsportalapp.herokuapp.com/home] but found [https://dsportalapp.herokuapp.com/]">
      <![CDATA[java.lang.AssertionError:  User was not redirected to the Home page. expected [https://dsportalapp.herokuapp.com/home] but found [https://dsportalapp.herokuapp.com/]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.LoginStepDef.the_user_should_be_redirected_to_the_home_page(LoginStepDef.java:156)
at ✽.the user should be redirected to the home page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/datastructure.feature:48)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="224.884"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="201.61"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="217.111">
    <failure type="java.lang.AssertionError" message="No error message displayed. expected [false] but found [true]">
      <![CDATA[java.lang.AssertionError: No error message displayed. expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertFalse(Assert.java:67)
at com.dsalgo.automation.stepdefinations.GraphStepDef.the_user_should_see_error_message_for_graph(GraphStepDef.java:88)
at ✽.the user should see an error message for graph(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/graph.feature:26)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="214.111"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="236.146"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="244.937"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="206.892"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="218.398"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="58.584">
    <failure type="org.openqa.selenium.WebDriverException" message="Timed out waiting for driver server to shutdown.
Build info: version: &amp;apos;4.31.0&amp;apos;, revision: &amp;apos;1ef9f18787*&amp;apos;
System info: os.name: &amp;apos;Mac OS X&amp;apos;, os.arch: &amp;apos;x86_64&amp;apos;, os.version: &amp;apos;14.6.1&amp;apos;, java.version: &amp;apos;23&amp;apos;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [f51a1c564c4abb73bff9aeaeca5e4a70, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54644}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54644/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: f51a1c564c4abb73bff9aeaeca5e4a70">
      <![CDATA[org.openqa.selenium.WebDriverException: Timed out waiting for driver server to shutdown.
Build info: version: '4.31.0', revision: '1ef9f18787*'
System info: os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '14.6.1', java.version: '23'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [f51a1c564c4abb73bff9aeaeca5e4a70, quit {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.96, chrome: {chromedriverVersion: 135.0.7049.95 (de2eb485a195..., userDataDir: /var/folders/c5/dbmv4z8n69j...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54644}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54644/devtoo..., se:cdpVersion: 135.0.7049.96, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: f51a1c564c4abb73bff9aeaeca5e4a70
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:312)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:161)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:618)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:622)
at org.openqa.selenium.remote.RemoteWebDriver.quit(RemoteWebDriver.java:446)
at org.openqa.selenium.chromium.ChromiumDriver.quit(ChromiumDriver.java:418)
at com.dsalgo.automation.driver.DriverFactory.quitDriver(DriverFactory.java:60)
at com.dsalgo.automation.hooks.Hooks.tearDownFreshBrowser(Hooks.java:50)
Caused by: org.openqa.selenium.net.UrlChecker$TimeoutException: Timed out waiting for http://localhost:8220/shutdown to become unavailable after 3141 ms
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:151)
at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:303)
... 8 more
Caused by: java.util.concurrent.TimeoutException
at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
at org.openqa.selenium.net.UrlChecker.waitUntilUnavailable(UrlChecker.java:144)
... 9 more
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="64.837"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="72.566"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="73.072"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="73.072"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="75.607">
    <failure type="java.lang.AssertionError" message="Expected &amp;apos;Data Structure Introduction&amp;apos; to be visible, but it was not. expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Expected 'Data Structure Introduction' to be visible, but it was not. expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertTrue(Assert.java:45)
at com.dsalgo.automation.stepdefinations.HomeStepDef.the_user_is_not_able_to_see_data_structure_introduction_in_the_dropdown(HomeStepDef.java:68)
at ✽.the user is not able to see Data Structure Introduction in the dropdown(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/home.feature:15)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="144.737">
    <failure type="java.lang.AssertionError" message=" User was not redirected to the Home page. expected [https://dsportalapp.herokuapp.com/home] but found [https://dsportalapp.herokuapp.com/]">
      <![CDATA[java.lang.AssertionError:  User was not redirected to the Home page. expected [https://dsportalapp.herokuapp.com/home] but found [https://dsportalapp.herokuapp.com/]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.LoginStepDef.the_user_should_be_redirected_to_the_home_page(LoginStepDef.java:156)
at ✽.the user should be redirected to the home page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/graph.feature:53)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="110.742">
    <failure type="java.lang.AssertionError" message="Overlap detected between paragraph and table/image! expected [false] but found [true]">
      <![CDATA[java.lang.AssertionError: Overlap detected between paragraph and table/image! expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertFalse(Assert.java:67)
at com.dsalgo.automation.stepdefinations.GraphStepDef.the_paragraph_should_not_overlap_with_matrix_table(GraphStepDef.java:151)
at ✽.the paragraph should not overlap with the matrix table(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/graph.feature:63)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="122.213">
    <failure type="java.lang.AssertionError" message="The Practice Questions page is blank.No content Found expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: The Practice Questions page is blank.No content Found expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertTrue(Assert.java:45)
at com.dsalgo.automation.stepdefinations.DataStructureStepDef.user_is_redirected_to_blank_practice_questions_page(DataStructureStepDef.java:157)
at ✽.the user is redirected to the blank page of "graph" "Practice Questions"(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/graph.feature:58)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="181.009"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="73.335"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="81.582"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="70.894"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="87.299"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="73.341"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="73.357"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="93.217"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="81.926"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="81.923"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="92.936"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="68.004"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="68.014"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="68.573"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="74.517"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="95.872">
    <failure type="java.lang.AssertionError" message=" Missing menu item: Data Structure Introduction expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError:  Missing menu item: Data Structure Introduction expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertTrue(Assert.java:45)
at com.dsalgo.automation.stepdefinations.HomeStepDef.dropdown_submenus_should_match_expected_items(HomeStepDef.java:132)
at ✽.the dropdown submenus should match the expected items from "dropdown" sheet(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/home.feature:85)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="100.199"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="75.631"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="86.553"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="75.715"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="87.219"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="69.937"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="101.035"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="90.455"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="106.604"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="65.122"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="69.59"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="34.695"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="36.083">
    <failure type="java.lang.AssertionError" message=" User was not redirected to the Home page. expected [https://dsportalapp.herokuapp.com/home] but found [https://dsportalapp.herokuapp.com/]">
      <![CDATA[java.lang.AssertionError:  User was not redirected to the Home page. expected [https://dsportalapp.herokuapp.com/home] but found [https://dsportalapp.herokuapp.com/]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at com.dsalgo.automation.stepdefinations.LoginStepDef.the_user_should_be_redirected_to_the_home_page(LoginStepDef.java:156)
at ✽.the user should be redirected to the home page(file:///Users/rajitha/DsAlgo/DSAlgo_March2025/src/test/resources/features/login.feature:41)
]]>
    </failure>
  </testcase> <!-- runScenario -->
</testsuite> <!-- Default test -->
